### PR TITLE
Use OpenSearch 2.3 by default instead of Elasticsearch 7.10, but provide compatibility for both

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** 

- https://github.com/stac-utils/stac-server/issues/118

**Proposed Changes:**

1. Update default database to use OpenSearch 2.3
2. Provide compatibility mode for Elasticsearch 7.10
3. Extensively document how to upgrade a stac-server CloudFormation Stack to the next version.

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/stac-utils/stac-server/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: opensearch/config/opensearch.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Cluster Settings
+cluster.name: stac-cluster
+node.name: es01
+network.host: 0.0.0.0
```
Comment (marchuffnagle) [inline opensearch/config/opensearch.yml:4]: We may want to make this `127.0.0.1` so that the service isn't available to everyone else in the coffee shop.

Comment (marchuffnagle) [reply opensearch/config/opensearch.yml:4]: Looking at how docker-compose is using this, I think it should be fine.

File: README.md [hunk 2]
```diff
@@ -95,13 +98,36 @@ apiLambda --> elasticsearch
...
 #### Elasticsearch upgrade from 7.9 to 7.10
 
-The Serverless Framework supports provisioning AWS resources, but it does not support updating existing resources. In 0.4, the default Elasticsearch version has been updated from 7.9 to 7.10. Continuing to use 7.9 should not cause any problems, but it recommended that you manually upgrade to 7.10 by going to [AWS Console - Amazon OpenSearch Service](https://console.aws.amazon.com/esv3/home), choosing the Elasticsearch domain used by your stac-server deployment (e.g., stac-server-{stage}-es), choose Upgrade from the Actions menu, and then upgrade to Elasticsearch 7.10.
+The Serverless Framework supports provisioning AWS resources, but it does not support updating existing resources. In 0.4, the default Elasticsearch version has been updated from 7.9 to 7.10. Continuing to use 7.9 should not cause any problems, but it recommended that you manually upgrade to 7.10 by going to [AWS Console - Amazon OpenSearch Service](https://console.aws.amazon.com/esv3/home), choosing the Elasticsearch domain used by your stac-server deployment (e.g., stac-server-{stage}-os), choose Upgrade from the Actions menu, and then upgrade to Elasticsearch 7.10.
```
Comment (marchuffnagle) [inline README.md:130]: Is this change accurate if someone is upgrading from v0.3.x to v0.4.x? It looks like the `-es` to `-os` change would be in v0.4.0 to v0.5.0?

Comment (philvarner) [reply README.md:130]: oh, I didn't look closely enough about where that change was when I made it. will revert.

**Comment by philvarner** [comment]
@matthewhanson @marchuffnagle this is ready for another review, thanks

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**